httpclient5/src/main/java/org/apache/hc/client5/http/impl/sync/HttpClientBuilder.java

```diff
@@ -888,6 +898,10 @@ public void close() throws IOException {
             closeablesCopy.add(connManagerCopy);
         }
 
+        if (tracer != null) {
+            execChain = new TracingExec(execChain, tracer, DefaultSpanManager.getInstance());
```

Comment:
some hook to do the same is the key. I also noticed the redirect problem (which could be ok as in okhttp we define an application span which makes a child for each network attempt)